arch/riscv/core/switch.S

```diff
@@ -14,6 +14,13 @@
 
 /* Convenience macros for loading/storing register states. */
 
+#if defined(CONFIG_RISCV_ISA_RV32E)
```

Comment:
Why not do this define just after `s1`, as everything is the same for the first 4 registers?

Reply:
ok

Reply:
actually after a second thought this is not possible since we cannot have an `ifdef` inside a `define`